README.md

```diff
@@ -46,35 +46,88 @@ application you can leave away the `poetry run` part of the commands given below
 
 1. To launch the program run `poetry run python -m app_webcam` in the root of this repository
 2. Enjoy
-3. Press `q` to terminate the program
+3. Press `q` or simply click the close button to terminate the program
 
-## Architecture Overview
+## Component Diagram
+
+```mermaid
+    graph TD
+        
+          IP
+          
+          PF
+          
+          subgraph Output Processes
+            TTS --> TTS_ON[TTS Online]
+            TTS --> TTS_OFF[TTS Offline]
+            
+            PL --> CSV_L[CSV Logger]
+            PL --> CSL[Console Logger]
+            
+            VIS ~~~ END:::hidden
+          end
+          
+          subgraph External Libraries
+              CV2
+              Numpy
+              Datetime
+              CSV
+              Pyttsx3
+              Gtts
+              Pygame
+          end
+          
+          APP_IMG --> APP
+          APP_WEB --> APP
+          
+          APP --> IP --> CV2
+          IP --> Numpy
+          
+          APP --> PF --> IP
```

Comment:
this seems wrong: the pattern filter doesn't use the image processor.

the import is only there for the `Pattern` class (only used for the typing hints) which we could arguably move into an own source file to make things clearer (if you want to do so you, you can just create a new file and then use the "refactor > move" method on `class Pattern` to move it to a new file). `Pattern` just started out in `image_proc.py` when we received it and we never moved it.

the same is also true for your links from the pattern logger, TTS and visualizer modules to the image processor

Reply:
good idea,  but let's do it in another commit